Move Secret Generation Strategies extension

baijum opened the pull request:
Based on the discussion in #158 and #156, this extension can become a
separate standard by itself. Move the extension into a separate file.

scothis commented:
Since this is our last "extension" should we also remove the whole section?

nebhale commented on README.md:
```diff
@@ -628,218 +623,3 @@ rules: [] # The control plane automatically fills in the rules
 # Extensions
```
Let's remove the entire Extensions section.

baijum replied:
Done. 1c0356458a54f4c896a6de8674f77995b0472a8c

nebhale reviewed (changes requested):
Also, let's make sure there's a concrete issue opened to ensure we don't lose the functionality, at least in an implementation and possibly as a separate spec.

baijum commented:
> Also, let's make sure there's a concrete issue opened to ensure we don't lose the functionality, at least in an implementation and possibly as a separate spec.

I have added the current extension text verbatim to this repository:
https://github.com/secret-generator/spec

baijum commented:
> > Also, let's make sure there's a concrete issue opened to ensure we don't lose the functionality, at least in an implementation and possibly as a separate spec.
> 
> I have added the current extension text verbatim to this repository:
> https://github.com/secret-generator/spec

Based on today's call. I have moved the extension into a separate file.

scothis commented on secret-generation.md:
```diff
@@ -0,0 +1,214 @@
+# Secret Generator Extension for Service Binding
```
it needs to be clear that this content is not a required part of the spec

baijum replied:
We can add a note. I need some help with phrasing it. Here is my attempt:

>Note: This spec is dependent on "Service Binding Specification for Kubernetes", but there is no direct association -- neither containment nor aggregation.

baijum replied:
One more sentence pointing to the release cycle:

> Note: This spec is dependent on "Service Binding Specification for Kubernetes", but there is no direct association -- neither containment nor aggregation. Both specs are going to have their independent release cycles.

scothis replied:
What if we moved it into a directory named "extensions" and added a README.md that contained [this paragraph](https://github.com/k8s-service-bindings/spec/pull/161/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L630)?

scothis replied:
By and large, this is the issue with keeping placeholder content in the repo. It becomes murky as to what it means and why it exists.

baijum replied:
- Created an "extensions" directory and moved the Secret Generation extension inside
- Created "extensions/README.md" with a brief introduction and a table of extensions (No., Title, and Status columns)
- Updated the extension with a brief introduction and status section

baijum commented:
Notes from yesterday's meeting:

* Evolved to a separate file
* Still needs a full review of content
* Think about how to display sub-specs (CloudEvents)

baijum commented:
> * Think about how to display sub-specs (CloudEvents)

I have created a extensions list in the [extensions/README.md](
https://github.com/k8s-service-bindings/spec/pull/161/files#diff-a4fc01cb110b1ca5c89a7624b5c8a4a455f52bd40eddfc554632625665b61ae4)

scothis reviewed (approved):
The move looks good, I'm not in a place to judge the content of the extension itself

arthurdm reviewed (approved):
LGTM
